File: week1/01-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        last_seen = {v:k for k, v in enumerate(nums)}
        for i, num in enumerate(nums):
            maybe_j = last_seen.get(target - num)
            if maybe_j and maybe_j != i:
                return [i, maybe_j]

File: week1/01-two-sum/two-sum.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> last_seen;
        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            const auto &it = last_seen.find(target - num);
            if (it != last_seen.end()) {
                return {it->second, i};
            }
            last_seen[num] = i;
        }
        return {};
    }
};